wpilibc/src/main/native/cpp/DriverStation.cpp

```diff
@@ -26,41 +28,70 @@
 #include "frc/WPIErrors.h"
 
 namespace frc {
+// A simple class which caches the previous value written to an NT entry
+// Used to prevent redundant, repeated writes of the same value
+template <class T>
+class MatchDataSenderEntry {
+ public:
+  MatchDataSenderEntry(std::shared_ptr<nt::NetworkTable> table,
+                       const std::string& key, const T& initialVal) {
+    static_assert(std::is_convertible<decltype(initialVal), bool>() ||
+                      std::is_convertible<decltype(initialVal), double>() ||
+                      std::is_convertible<decltype(initialVal), std::string>(),
+                  "Invalid type for MatchDataSenderEntry - must be convertable "
+                  "to bool, double or std::string");
+
+    ntEntry = table->GetEntry(key);
+    if constexpr (std::is_convertible<decltype(initialVal), bool>()) {
+      ntEntry.ForceSetBoolean(initialVal);
+    } else if constexpr (std::is_convertible<decltype(initialVal), double>()) {
+      ntEntry.ForceSetDouble(initialVal);
+    } else if constexpr (std::is_convertible<decltype(initialVal),
+                                             std::string>()) {
+      ntEntry.ForceSetString(initialVal);
+    }
+    prevVal = initialVal;
+  }
+  void Set(const T& val) {
```

Comment:
For consistency, add a blank line before Set().

Reply:
Fixed.